pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -233,6 +233,19 @@ func (r *ReconcileCephBlockPoolRadosNamespace) reconcile(request reconcile.Reque
 		return reconcile.Result{}, nil
 	}
 
+	// Build the NamespacedName to fetch the CephBlockPool and make sure it exists, if not we cannot
+	// create the rados namespace
+	cephBlockPool := &cephv1.CephBlockPool{}
+	pool := cephBlockPoolRadosNamespace.Spec.BlockPoolName
+	cephBlockPoolNamespacedName := types.NamespacedName{Name: pool, Namespace: request.Namespace}
+
+	if csi.EnableCSIOperator() || cephCluster.Spec.External.Enable {
+		err = csi.CreateUpdateClientProfileRadosNamespace(r.clusterInfo.Context, r.client, r.clusterInfo, cephBlockPoolNamespacedName, buildClusterID(cephBlockPoolRadosNamespace), cephCluster.Name)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to create ceph csi-op config CR for RadosNamespace")
+		}
+	}
+
 	if cephCluster.Spec.External.Enable {
 		logger.Debugf("external rados namespace %q creation is not supported, create it manually, the controller will assume it's there", namespacedName)
```

Comment:
And this is contradictory for rados namespaces as well

Reply:
removed the condition